### PR TITLE
update oauth2-proxy readme

### PR DESCRIPTION
**Pull request by juliusvonkohout**
@tarekabouzeid please /lgtm

### CODE REVIEW

File: common/oauth2-proxy/README.md [hunk 1]
```diff
@@ -76,12 +76,111 @@ You can adjust OAuth2 Proxy to directly connect to your own IDP(Identity Provide
...
+      claim: groups
+
+    # We explicitly set `fromHeaders` to ensure that the JWT is only extracted from the `Authorization` header.
+    # This is because we exclude requests that have an `Authorization` header from oauth2-proxy.
+    fromHeaders:
+    - name: Authorization
+      prefix: "Bearer "
+```
+
+You can also add more RequestAuthentication to support other issuers as for eyample for M2M acces from github actions as explained in the root level Readme.md.
```
Comment (tarekabouzeid) [inline common/oauth2-proxy/README.md:180]:
```suggestion
You can also add more RequestAuthentication to support other issuers as for example for M2M access from github actions as explained in the root level Readme.md.
```

File: common/oauth2-proxy/README.md [hunk 2]
```diff
@@ -76,12 +76,111 @@ You can adjust OAuth2 Proxy to directly connect to your own IDP(Identity Provide
...
+This feature is useful when you need to integrate Kubeflow with you current CI/CD platform (GitHub Actions, Jenkins) via machine-to-machine authentication.
```
Comment (tarekabouzeid) [inline common/oauth2-proxy/README.md:181]:
```suggestion
This feature is useful when you need to integrate Kubeflow with your current CI/CD platform (GitHub Actions, Jenkins) via machine-to-machine authentication.
```

**Comment by juliusvonkohout** [comment]
The test ran to slow and i restarted Deploy and test Kubeflow Pipelines manifests with seaweedfs and m2m auth in KinD / build

**Comment by juliusvonkohout** [comment]
and its green

**Pull request merged**